server/app/models/company.rb

```diff
@@ -0,0 +1,5 @@
+class Company < ApplicationRecord
+    has_many :users
+    has_many :allowlist_domains
+    has_many :allowlist_emails
```

Comment:
nit: I think it would be cleaner if the Company model held on to the user type rather than the AllowlistEmail or AllowlistDomain. But either way is fine with me.